frame/clearing-house/src/lib.rs

```diff
@@ -540,6 +549,20 @@ pub mod pallet {
 				price_spread.checked_mul(&period_adjustment).ok_or(ArithmeticError::Underflow)?;
 			Ok(rate)
 		}
+
+		fn funding_owed(
+			market: &Self::Market,
+			position: &Self::Position,
+		) -> Result<Self::Decimal, DispatchError> {
+			let cum_funding_delta = market
+				.cum_funding_rate
+				.checked_sub(&position.last_cum_funding)
+				.ok_or(ArithmeticError::Underflow)?;
+			let payment = cum_funding_delta
+				.checked_mul(&position.base_asset_amount)
+				.ok_or(ArithmeticError::Overflow)?;
```

Comment:
For what I understood about `cum_funding_delta` and `base_asset_amount` looking the tests `funding_owed_is_nonzero_iff_cum_rates_not_equal` and `funding_owed_query_leaves_storage_intact` these two variables could assume negative values, right?

If that's the case, there could be a case where `cum_funding_delta * position.base_asset_amount` produces a value smaller than Decimal::MIN, in which case the error would technically be an _underflow_ rather than an _overflow_, right?

Take this test for example:

```rust
#[test]
fn funding_owed_underflow() {
	ExtBuilder::default().build().execute_with(|| {
		let market = Market {
			vamm_id: 0,
			asset_id: 0,
			margin_ratio_initial: FixedI128::from_float(0f64),
			margin_ratio_maintenance: FixedI128::from_float(0f64),
			cum_funding_rate: FixedI128::from_float(3f64),
			funding_rate_ts: 0,
			funding_frequency: 0,
			funding_period: 0,
		};

		let position = Position {
			market_id: 0,
			base_asset_amount: FixedI128::from_inner(i128::MIN),
			quote_asset_notional_amount: FixedI128::from_float(0f64),
			last_cum_funding: FixedI128::from_float(1f64),
		};

		let result = <TestPallet as Instruments>::funding_owed(&market, &position);
		dbg!(position, result);

		assert_ok!(result);
	})
}
```

Reply:
Yeah, you're right. I'll take a look at `checked_mul` to see if there's a way to disambiguate the two errors and raise the appropriate `ArithmeticError`

Reply:
Maybe you could just check the sign of the variables and make a `match` in case of errors?